conf/schemas/dataSets.schema.js

```diff
@@ -0,0 +1,79 @@
+db.runCommand({
+  collMod: "dataSets",
+  validator: {
+    $and: [
+      {
+        dataStoreInfo: { $type: "object", $exists: true },
+      },
+      { "dataStoreInfo.name": { $regex: "^[A-Za-z0-9-_.]+$", $exists: true } },
+      {
+        "dataStoreInfo.url": {
+          $regex: "^https?:/{2}[a-z0-9.-]+(.[a-z]{2,3})?(:[0-9]+)?$",
+          $exists: true,
+        },
+      },
+      { "dataStoreInfo.typ": { $in: ["webknossos-store", "ndstore"], $exists: true } },
+      {
+        $or: [
+          { "dataStoreInfo.accessToken": { $type: "string" } },
+          { "dataStoreInfo.accessToken": { $exists: false } },
+        ],
+      },
+      {
+        dataSource: { $type: "object", $exists: true },
+      },
+      {
+        "dataSource.id": { $type: "object", $exists: true },
+      },
+      {
+        "dataSource.id.name": { $regex: "^[A-Za-z0-9-_.]+$", $exists: true },
+      },
+      {
+        "dataSource.id.team": { $type: "string", $exists: true },
+      },
+      {
+        $or: [
+          { "dataSource.dataLayers": { $type: "object" } },
+          { "dataSource.dataLayers": { $exists: false } },
+          { "dataSource.dataLayers": { $size: 0 } },
```

Comment:
Can we also check the properties of each dataLayer? By setting `$type: { <some nested properties> }`? If this is possible, we should do that.